### ISSUE TITLE
Close instances of dart.core.sink doesn't recognize close when using this

### ISSUE THREAD

**Issue by grandpa-guru** [dart-lang/linter#1446]
**Describe the issue**
When closing the `Sink` with accessing to it through `this` warning appears that the `Sink` is not closed even tho it is.

**To Reproduce**
1. Add a property of type `Sink` in your class
2. Close the sink using `this.sink.close()`
3. Warning `Close instances of dart.core.Sink` apperas
4. Remove `this`
5. No more warnings

**Expected behavior**
There should be no warnings when closing the `dart.core.Sink` with or without accessing to it through `this`.

Is this working as expected? To me it looks it's not.

**Comment by pq** [comment]
Thanks for the report @grandpa-guru!   It looks like this is easy to reproduce:

```dart
class B0 {
  IOSink _sinkB0; // <= Wrongly flagged.
  void init(filename) {
    _sinkB0 = new File(filename).openWrite();
  }

  void dispose(filename) {
    // ignore: unnecessary_this
    this._sinkB0.close();
  }
}
```

We'll take a look at a fix. 👍

**Comment by grandpa-guru** [comment]
Any progress on this?

**Comment by grandpa-guru** [comment]
> Thanks for the report @grandpa-guru! It looks like this is easy to reproduce:
> 
> ```dart
> class B0 {
>   IOSink _sinkB0; // <= Wrongly flagged.
>   void init(filename) {
>     _sinkB0 = new File(filename).openWrite();
>   }
> 
>   void dispose(filename) {
>     // ignore: unnecessary_this
>     this._sinkB0.close();
>   }
> }
> ```
> 
> We'll take a look at a fix. 👍

Any progress on this?

**Comment by pq** [comment]
Sorry @grandpa-guru.  Not yet.  I'll try and take a look this week. 🤞

**Comment by javiermrz** [comment]
This is still happening to me:
 ```dart
class PLCVarControlScreen extends StatefulWidget {
  @override
  _PLCVarControlScreenState createState() => _PLCVarControlScreenState();
}

class _PLCVarControlScreenState extends State<PLCVarControlScreen> {
// This gives warning: Close instances of `dart.core.Sink`.
  StreamController<PLC> _plcStreamController;

 @override
   void dispose() {
     super.dispose();
     this._plcStreamController.close();
   }
...
}
```

Any progress? :)

**Comment by pq** [comment]
Sorry, no significant progress.  I took a look tonight but it's a little hairy and I need to get back a little more context.

Sorry for the delay!

**Comment by LukaGiorgadze** [comment]
@pq , Hey, any news? 🗞️

**Comment by pq** [comment]
Hey @LukaGiorgadze.  Sorry... no.   This keeps falling off.  I'll try and bump it back up.

Thanks for your patience!
